components/match2/wikis/brawlstars/match_summary.lua

```diff
@@ -0,0 +1,365 @@
+		end
+	end
+
+	match.links.lrthread = match.lrthread
+	match.links.vod = match.vod
+	if not Table.isEmpty(vods) or not Table.isEmpty(match.links) then
+		local footer = MatchSummary.Footer()
+
+		-- Game Vods
+		for index, vod in pairs(vods) do
+			footer:addElement(VodLink.display{
+				gamenum = index,
+				vod = vod,
+				source = vod.url
+			})
+		end
+
+		-- Match Vod + other links
+		local buildLink = function (linkType, link)
+			local linkData = _LINK_DATA[linkType]
+			if not linkData then
+				mw.log('linkType "' .. linkType .. '" is not supported by Module:MatchSummary')
+			else
+				return '[[' .. linkData.icon .. '|link=' .. link .. '|15px|' .. linkData.text .. ']]'
+			end
+		end
+
+		for linkType, link in pairs(match.links) do
+			footer:addElement(buildLink(linkType,link))
+		end
+
+		matchSummary:footer(footer)
+	end
+
+	return matchSummary:create()
+end
+
+
+function CustomMatchSummary._createHeader(match)
+	local header = MatchSummary.Header()
+
+	header:leftOpponent(header:createOpponent(match.opponents[1], 'left'))
+	      :leftScore(header:createScore(match.opponents[1]))
+	      :rightScore(header:createScore(match.opponents[2]))
+	      :rightOpponent(header:createOpponent(match.opponents[2], 'right'))
+
+	return header
+end
+
+function CustomMatchSummary._createBody(match)
+	local body = MatchSummary.Body()
+
+	if match.dateIsExact or (match.date ~= _EPOCH_TIME_EXTENDED and match.date ~= _EPOCH_TIME) then
+		-- dateIsExact means we have both date and time. Show countdown
+		-- if match is not epoch=0, we have a date, so display the date
+		body:addRow(MatchSummary.Row():addElement(
+			DisplayHelper.MatchCountdownBlock(match)
+		))
+	end
+
+	-- Iterate each map
+	for _, game in ipairs(match.games) do
+		if game.map then
+			body:addRow(CustomMatchSummary._createMapRow(game))
+		end
+	end
+
+	-- Add Match MVP(s)
+	local mvpInput = match.extradata.mvp
+	if mvpInput then
+		local mvpData = mw.text.split(mvpInput or '', ',')
+		if String.isNotEmpty(mvpData[1]) then
+			local mvp = MatchSummary.Mvp()
+			for _, player in ipairs(mvpData) do
+				if String.isNotEmpty(player) then
+					mvp:addPlayer(player)
+				end
+			end
+
+			body:addRow(mvp)
+		end
+	end
+
+	-- Pre-Process Brawler picks
+	local showGamePicks = {}
+	for gameIndex, game in ipairs(match.games) do
+		local pickData = {{}, {}}
+		local numberOfPicks = game.extradata.maximumpickindex
+		local participants = game.participants
+		for index = 1, numberOfPicks do
+			if not Table.isEmpty(participants['1_' .. index]) then
+				pickData[1][index] = participants['1_' .. index].brawler
+			end
+			if not Table.isEmpty(participants['2_' .. index]) then
+				pickData[2][index] = participants['2_' .. index].brawler
+			end
+		end
+
+		if numberOfPicks > 0 then
+			pickData.numberOfPicks = numberOfPicks
+			showGamePicks[gameIndex] = pickData
+		end
+	end
+
+	-- Add the Brawler pics
+	if not Table.isEmpty(showGamePicks) then
+		local brawler = Brawler('Picks')
```

Comment:
If the Brawler class is initiated with Picks/Bans, why is the `isBan` paramter needed for the row?

Reply:
tbh i was tired, of course you are right